src/ffi/ext.rs

```diff
@@ -238,14 +238,14 @@ extern "C" {
         request: c_uint,
         reply: *mut *mut c_void,
         e: *mut *mut xcb_generic_error_t,
-    ) -> *mut c_void;
+    ) -> *mut c_int;
 
     pub(crate) fn xcb_poll_for_reply64(
         c: *mut xcb_connection_t,
         request: u64,
         reply: *mut *mut c_void,
         e: *mut *mut xcb_generic_error_t,
-    ) -> *mut c_void;
+    ) -> *mut c_int;
```

Comment:
`xcb_poll_for_reply` and `xcb_poll_for_reply64` return `c_int`, not `c_void`. The `c_void`, which represents the returned reply, is what `*reply` is set to. The return value instead represents whether the reply has been received yet: `0` if the reply has not been received (i.e. `None`), `1` if it has (i.e. `Some`).

Reply:
@Antikyth can you correct the return type of both FFI functions to `c_int` (as opposed to `*mut c_int`) ?
The return type is not a pointer in the C API.
https://gitlab.freedesktop.org/xorg/lib/libxcb/-/blob/master/src/xcbext.h?ref_type=heads#L291

Reply:
> @Antikyth can you correct the return type of both FFI functions to `c_int` (as opposed to `*mut c_int`) ? The return type is not a pointer in the C API. https://gitlab.freedesktop.org/xorg/lib/libxcb/-/blob/master/src/xcbext.h?ref_type=heads#L291

Fixed it now (my apologies, I forgot about it - I wasn't able to when I saw the other message last week)